WaveSim.m

```diff
@@ -57,6 +57,43 @@
                 E = f_wiggle(E, conj(wig.gx), conj(wig.gy), conj(wig.gz));
             end
         end
+
+        function state = run_algorithm(obj, state)
```

Comment:
[discuss in next group meeting?]
Worth trying to merge the new run_algorithm function here into the preexisting run_algorithm function in WaveSimBase.